### PR TITLE
Updated all reload modules related tests for backport

### PR DESCRIPTION
**Pull request by ZiyadAtiyyeh**
Updates for when using asap mlnx ofed machine in tests:
test-ecmp-devlink-multipath.sh
test-eswitch-add-flows-during-reload.sh
test-eswitch-add-flows-during-reload.sh
test-reload-modules-different-state.sh

### CODE REVIEW

File: test-ecmp-devlink-multipath.sh [hunk 1]
```diff
@@ -38,7 +38,12 @@ function enable_disable_multipath() {
     if [ -z "$mode" ]; then
         mode='X'
     fi
-    test $mode = "enable" || err "Expected multipath mode enabled but got $mode"
+
+    if [ "$devlink_compat" = 1 ]; then
+        test $mode = "enabled" || err "Expected multipath mode enabled but got $mode"
+    else
+	test $mode = "enable" || err "Expected multipath mode enabled but got $mode"
+    fi
```
Comment (roidayan) [inline test-ecmp-devlink-multipath.sh]: spaces are not aligned here.

also in the commit title you have unneeded space.

File: test-eswitch-del-flows-during-reload.sh [hunk 2]
```diff
@@ -53,8 +53,13 @@ function del_rules() {
...
 function reload_modules() {
-    modprobe -r mlx5_ib mlx5_core devlink || fail "Failed to unload modules"
-    modprobe -a devlink mlx5_core mlx5_ib || fail "Failed to load modules"
+    if [ "$devlink_compat" = 1 ]; then
+	service openibd force-restart
+    else
+        modprobe -r mlx5_ib mlx5_core devlink || fail "Failed to unload modules"
+        modprobe -a devlink mlx5_core mlx5_ib || fail "Failed to load modules"
+    fi
+
```
Comment (roidayan) [inline test-eswitch-del-flows-during-reload.sh]: doesnt this fail in the backport because cls_flower is busy?

Comment (ZiyadAtiyyeh) [reply test-eswitch-del-flows-during-reload.sh]: No, with force-restart, it did not fail and works fine in this case.

File: test-ecmp-devlink-multipath.sh (hunk 1 above)
Comment (ZiyadAtiyyeh) [reply test-ecmp-devlink-multipath.sh]: I will fix this and push the new changes.

Comment (ZiyadAtiyyeh) [reply test-ecmp-devlink-multipath.sh]: Done.

**Pull request merged**